### DISCUSSION TITLE
Can't compile Assembly aarch64. Need Help !

### DISCUSSION THREAD

**Discussion by Craglitch** [termux/termux-packages#9376] [Q&A]
Hi community, i have problem with my termux when i compile assembly it's give me error message.

```

 .text

.global _start

_start:
        MOV R0, #86
        MOV R1, #29

SWI 0

```

Compile output using ``as`` command.

```
ASM.s: Assembler messages:
ASM.s:6: Error: operand 1 must be an integer register -- `mov R0,#86'
ASM.s:7: Error: operand 1 must be an integer register -- `mov R1,#29'
ASM.s:9: Error: unknown mnemonic `swi' -- `swi 0'
```

My lscpu output.
![IMG_20220312_132505](https://user-images.githubusercontent.com/83074053/158005124-4e2e5f9d-4dce-46b2-bda0-0b777818f6c5.jpg)

Application version:
0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.astra.in.ua/apt/termux-main/ stable main
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.org/apt/termux-x11 x11 main
Updatable packages:
apt/stable 2.4.1 aarch64 [upgradable from: 2.3.15-1]
cmake/stable 3.22.3 aarch64 [upgradable from: 3.22.1-1]
Android version:
8.1.0
Kernel build information:
Linux localhost 4.9.77+ #2 SMP PREEMPT Fri Mar 19 20:17:40 CST 2021 aarch64 Android
Device manufacturer:
OPPO
Device model:
CPH1909
 
Please help, thanks.

**Comment by ar664** [comment]
Could possibly be incorrect syntax.
The following code compiles with the current as:
https://github.com/odzhan/shellcode/tree/master/os/linux/arm64

**Comment by licy183** [comment]
It seems that your asm file is an arm32 one.
Currently, `binutils` in the main repo doen't support cross compile, so you can try to use `as` on an arm32 device. If you really need to compile it on aarch64 device, you can try to use `clang --target=arm`, but it will fail due to the linker error.
Anyway, `llvm-as` is not a GNU `as` replacement. Actually, they are totally different. But `binutils-is-llvm` wrongly symlinked `llvm-as` to `as`. I will try to fix it tomorrow.
